### DISCUSSION TITLE
Grafana only when in Terminal Mode

### DISCUSSION THREAD

**Discussion by Cube3SGM** [David00/rpi-power-monitor#143] [General]
Hello,

I have had a data corruption on the SD Card stopping RPi-PM from loading. I had an extra SD Card for back-ups, but do not know how to access this.

Anyway, no problem I downloaded the RPi-PM image, set the correct parameters in config.toml. Check status of power-monitor.service everything looks ok.  But seeing no data in Grafana. It doesn't say "No Data", the readings are static, the time is passing but no graphs or shaded area underneath the plots.   If I run

python3 ~/rpi_power_monitor/rpi_power_monitor/power_monitor.py --mode terminal

I see the readings in Terminal and start to see data come into Grafana with a straight diagonal line from the old data across time to the new data point, As soon as I quit Terminal mode (ctl-C) Grafana stops receiving data.

Any help appreciated.
Regards Simon

**Comment by David00** [comment]
Hi Simon, can you check to see if the power monitor is started in the background as a service? Use:

```
sudo systemctl status power-monitor.service
```

If it's not running, start it with:

```
sudo systemctl start power-monitor.service
```

...and then check it again with the status command.

**Comment by Cube3SGM** [comment]
![Power-Monitor - Capture01](https://github.com/user-attachments/assets/22dba523-ed17-4a3b-998f-1ef5998a83ba)

As far as I can tell it is running fine!

Regards Simon

**Reply by David00** [reply]
Ah, I see what's happening. You're using a username of `raspberry` but the background service expects a username of `pi`.

You'll have to edit the service file to change the username and the path to reflect the different username.

Run the command `sudo nano /etc/systemd/system/power-monitor.service` to edit the service file, and replace all instances of `pi` with `raspberry`.

Specifically, these two lines:
```
User=raspberry
ExecStart=/usr/bin/python3 /home/raspberry/rpi_power_monitor/rpi_power_monitor/power_monitor.py
```

Then, save and close the file with `Ctrl-x`, then `y`, and reload the file with:  `sudo systemctl daemon-reload`

Finally, try to start the power monitor again with `sudo systemctl start power-monitor.service` and then check it with `sudo systemctl status power-monitor`.  Note that the line with "Active" should say active (running), like this:

```
● power-monitor.service - Raspberry Pi Power Monitor
   Loaded: loaded (/etc/systemd/system/power-monitor.service; enabled; vendor preset: enabled)
   Active: active (running) since Mon 2024-09-30 11:27:04 PDT; 2 weeks 0 days ago
```

**Comment by Cube3SGM** [comment]
![Power-Monitor - Capture02](https://github.com/user-attachments/assets/86926dc8-022f-4e7e-a0ea-87c86b161f2c)

User already pi

Thank you for helping David.

**Reply by David00** [reply]
It needs to be `raspberry` since that is the username you are using.

And the line containing ExecStart needs to point to /home/raspberry/ instead of /home/pi.

**Comment by Cube3SGM** [comment]
Sorry for the mis-understanding, all is working fine now, thank you.

Not sure if the Raspberry Pi Imager had a hand in changing the user to "raspberry", it's all I can think of.

Thanks again.
Simon
